src/core/xref.js

```diff
@@ -46,6 +46,17 @@ function createXrefMap(elems) {
           .replace(/!/g, "")
           .split(" ")
       : [];
+    if (datacite !== elem) {
+      // if element itself contains data-cite, we don't take inline context into account
+      const refs = elem.closest("section").querySelectorAll("a.bibref");
+      if (refs) {
+        const inlineContextSpecs = [...refs].map(el =>
+          el.textContent.toLowerCase().replace(/^!/, "")
```

Comment:
We should do the removal of the “!” when the inline references are converted. So, in core/inlines i think.